ymer/src/main/java/com/avanza/ymer/MongoDocumentCollection.java

```diff
@@ -118,12 +129,15 @@ public void update(Document newVersion) {
 	}
 
 	@Override
-	public void updateById(Object id, Map<String, Object> fieldsAndValuesToSet) {
-		fieldsAndValuesToSet.entrySet().stream().map(entry -> Updates.set(entry.getKey(), entry.getValue())).reduce(Updates::combine)
-				.ifPresent(updates -> {
-					UpdateResult updateResult = collection.updateOne(Filters.eq(id), updates);
-					idValidator.validateUpdatedExistingDocument("update", updateResult, new Document("_id", id));
-				});
+	public void updateAllPartial(List<Document> documents) {
```

Comment:
Could the implementation be simplified by accepting `Map<Object, Document> updatesById` as parameter instead? Maybe like this:
```java
List<UpdateOneModel<Document>> updates = updatesById.entrySet().stream()
		.map(e -> new UpdateOneModel<Document>(Filters.eq(e.getKey()), e.getValue()))
		.collect(toList());
```

Reply:
According to the documentation of `UpdateOneModel`
> update – a document describing the update, which may not be null. The update to apply must include only update operators.

Ie. the BSON contains a number of update operators, such as `$set`. 

The way this method is implemented now, you send in a list of documents containing the `_id` field and the fields with values that you want to set. `MongoDocumentCollection` then creates the `$set` operations. 

It cannot be a Document with fields to update. So simplifying this method would mean moving the logic of creating the update operations to the caller. I don't think that is the appropriate abstract level for the current use case.